src/shims/unix/linux/foreign_items.rs

```diff
@@ -191,6 +191,21 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.set_last_error(einval)?;
                 this.write_scalar(Scalar::from_i32(-1), dest)?;
             }
+            "sched_setaffinity" => {
+                // a no-op. The assumption is that application code should be able to recover from
+                // this operation failing with EINVAL, which in this case indicates that the
+                // affinity bit mask does not contain any processors that are physically on the
+                // system.
```

Comment:
I don't think this is a valid assumption; wouldn't application code using an API based on `sched_setaffinity` have just queried which processors are on the system, and thus be confident that this error case won't be hit?

Reply:
We have a a flag `-Zmiri-num-cpus`, which defaults to `1` (and I for one routinely set to large numbers). I'd prefer if the behavior of this shim were based on that flag.

Reply:
well, conveniently miri's `libc::sched_getaffinity` also always returns `EINVAL`, so yeah in proper code I guess  you'd never actually reach the `libc::sched_setaffinity`.

I think the real reason for this implementation is that miri doesn't want to simulate this function properly, so returning an error is the only thing that is left to be done.

Reply:
Ouch. I see your line of thinking now.